code/game/gamemodes/changeling/powers/tiny_prick.dm

```diff
@@ -66,60 +66,44 @@
+	button_icon_state = "sting_armblade"
+	sting_icon = "sting_armblade"
+	chemical_cost = 20
+	dna_cost = 2
+	genetic_damage = 15
+	max_genetic_damage = 25
 
-/datum/action/changeling/sting/transformation/can_sting(var/mob/user, var/mob/target)
-	if(!..())
-		return
-	if((HUSK in target.mutations) || (!ishuman(target)))
-		to_chat(user, "<span class='warning'>Our sting appears ineffective against its DNA.</span>")
-		return FALSE
-	if(ishuman(target))
-		var/mob/living/carbon/human/H = target
-		if(NO_DNA in H.dna.species.species_traits)
-			to_chat(user, "<span class='warning'>This won't work on a creature without DNA.</span>")
-			return FALSE
-	return TRUE
+/obj/item/melee/arm_blade/false
+	desc = "A grotesque mass of flesh that used to be your arm. Although it looks dangerous at first, you can tell it's actually quite dull and useless."
+	force = 5 //Basically as strong as a punch
+	fake = TRUE
 
-/datum/action/changeling/sting/transformation/sting_action(var/mob/user, var/mob/target)
-	add_attack_logs(user, target, "Transformation sting (changeling) (new identity is [selected_dna.real_name])")
-	if(issmall(target))
-		to_chat(user, "<span class='notice'>Our genes cry out as we sting [target.name]!</span>")
+/datum/action/changeling/sting/false_armblade/sting_action(var/mob/user, var/mob/living/carbon/target)
+	add_attack_logs(user, target, "Fake armblade sting (changeling)")
 
-	if(iscarbon(target) && (target.status_flags & CANWEAKEN))
-		var/mob/living/carbon/C = target
-		C.do_jitter_animation(500)
+	if(!target.drop_item())
+		to_chat(user, "<span class='warning'>The [target.get_active_hand()] is stuck to [target.p_their()] hand, we cannot grow a false armblade over it!</span>")
+		return
+	..()
 
-	target.visible_message("<span class='danger'>[target] begins to violenty convulse!</span>","<span class='userdanger'>You feel a tiny prick and a begin to uncontrollably convulse!</span>")
+	var/obj/item/melee/arm_blade/false/blade = new(target,1)
+	target.put_in_hands(blade)
+	target.visible_message("<span class='warning'>A grotesque blade forms around [target.name]\'s arm!</span>", "<span class='userdanger'>Your arm twists and mutates, transforming into a horrific monstrosity!</span>", "<span class='hear'>You hear organic matter ripping and tearing!</span>")
 
-	spawn(10)
-		transform_dna(target,selected_dna)//target is always human so no problem here
-	feedback_add_details("changeling_powers","TS")
+	addtimer(CALLBACK(src, /datum/action/changeling/sting/false_armblade/proc/remove_fake, target, blade), 600)
 	return TRUE
 
+
+/datum/action/changeling/sting/false_armblade/proc/remove_fake(mob/target, obj/item/melee/arm_blade/false/blade)
+	target.visible_message("<span class='warning'>With a sickening crunch, [target] reforms [target.p_their()] [blade.name] into an arm!</span>","<span class='userdanger'>Your arm twists and mutates... Back to normal, thank god.</span>",)
+	qdel(blade)
```

Comment:
Ensure this actually removes the blade from the equiped hand. Might have to unequip it first

Reply:
does work